CODE_OF_CONDUCT.md

```diff
@@ -0,0 +1,76 @@
+# Contributor Covenant Code of Conduct
+
+## Our Pledge
+
+In the interest of fostering an open and welcoming environment, we as
+contributors and maintainers pledge to making participation in our project and
+our community a harassment-free experience for everyone, regardless of age, body
+size, disability, ethnicity, sex characteristics, gender identity and expression,
+level of experience, education, socio-economic status, nationality, personal
+appearance, race, religion, or sexual identity and orientation.
+
+## Our Standards
+
+Examples of behavior that contributes to creating a positive environment
+include:
+
+* Using welcoming and inclusive language
+* Being respectful of differing viewpoints and experiences
+* Gracefully accepting constructive criticism
+* Focusing on what is best for the community
+* Showing empathy towards other community members
+
+Examples of unacceptable behavior by participants include:
+
+* The use of sexualized language or imagery and unwelcome sexual attention or
+ advances
+* Trolling, insulting/derogatory comments, and personal or political attacks
+* Public or private harassment
+* Publishing others' private information, such as a physical or electronic
+ address, without explicit permission
+* Other conduct which could reasonably be considered inappropriate in a
+ professional setting
+
```

Comment:
If you want to use the Contributor Covenant, I think several of these should have a clear "Good/Bad" example that specifically illustrates said behavior so there is less ambiguity.

This was very painful, but I went through some abusive instances in TSDX's history that occurred to multiple maintainers and paraphrased to make examples:
```md
* Snarky, sarcastic, condescending, or passive-aggressive behavior:
❌  Bad: "Will you fix this issue, or should I just not use this library?"
✅  Good: "I'd like to use this library, but this issue is a blocker for me right now."

❌  Bad: "How many more breaking bugs do you need?"
✅  Good: "These bugs have been very frustrating for me, so I would appreciate if changes respected SemVer more carefully and testing were a higher priority".

❌  Bad: "_Really_?"
✅  Good: "I believe this line doesn't match up with what you said, but correct me if I'm wrong"

* Trolling, insulting/derogatory comments, and personal or political attacks
❌  Bad: "You shouldn't have closed my issue, you're a jerk"
✅  Good: "I disagree that it should be closed and think this issue is substantial for these reasons [...]. Maybe this will be reconsidered down the line"

❌  Bad: "Who put _you_ in charge??"
✅  Good: "I disagree with this approach and think it could be improved in these ways [...]. Maybe this can be reconsidered on that basis"

❌  Bad: "You're uninformed and don't know what you're talking about. Your opinion is wrong. I'll take my chances with other libraries."
✅  Good: "Here are some examples that illustrate my point [...]. I disagree with this opinion and think the reverse trade-offs are better to make. I hope these examples can better inform the community's decisions, but I feel strongly enough about these trade-offs that I'll be exploring other approaches."
```

On the positive side, here's a few other examples:
```md
❌  Bad: "My word is final."
✅  Good: "Here's a few reasons why I don't think that is a good approach [...]. The examples and evidence here are not currently enough to sway my opinion, but if there is more community support, maybe it can be reconsidered then."

❌  Bad: "I'm not going to use that"
❌  Bad: "I don't like that"
❌  Bad: "That's a bad option"
✅  Good: "I don't agree with that option because of these trade-offs: [...]. If more positive trade-offs outweigh the cons, maybe I'll reconsider"
```